### PR TITLE
Fix oversights in Config

### PR DESCRIPTION
**Pull request by viliam-durina**
The name without the part after "@" is used to look up, however when it is not found, config with full name is inserted to the map.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/Config.java [hunk 1]
```diff
@@ -311,12 +311,16 @@ public Config setMapConfigs(Map<String, MapConfig> mapConfigs) {
 
     public CacheSimpleConfig findCacheConfig(String name) {
         name = getBaseName(name);
-        return lookupByPattern(configPatternMatcher, cacheConfigs, name);
+        final CacheSimpleConfig config = lookupByPattern(configPatternMatcher, cacheConfigs, name);
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/config/Config.java]: This change in `CacheSimpleConfig` lookup may alter our JCache implementation's behavior from throwing `CacheException` on `CacheManager.getCache(unknownCacheName)` to returning a `default`-configured `Cache`. There is already an issue for that (#10695) but it's a significant change and its consequences should be explored carefully, so I wouldn't do that in this PR.

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
verify

**Comment by viliam-durina** [comment]
verify

File: hazelcast/src/main/java/com/hazelcast/config/Config.java [hunk 2]
```diff
@@ -484,20 +484,28 @@ public Config setMapConfigs(Map<String, MapConfig> mapConfigs) {
...
+     * <p>
      * The name is matched by pattern to the configuration and by stripping the
      * partition ID qualifier from the given {@code name}.
+     * If there is no config found by the name, it will return the configuration
+     * with the name {@code default}.
      *
-     * @param name name of the cache config
-     * @return the cache configuration or {@code null} if none was found
+     * @param name name of the cardinality estimator config
+     * @return the cardinality estimator configuration
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/config/Config.java]: Something went wrong with rebasing here? this javadoc should be about cache configs.

File: hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigurationAwareConfig.java [hunk 3]
```diff
@@ -243,7 +243,7 @@ public Config setMapConfigs(Map<String, MapConfig> mapConfigs) {
     }
 
     @Override
-    public CacheSimpleConfig findCacheConfig(String name) {
+    public CacheSimpleConfig findCacheConfigOrNull(String name) {
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigurationAwareConfig.java]: Doesn't this change the behaviour of the `Config.findCacheConfig(name)` for a configuration obtained from a running hazelcast instance? @vbekiaris is this acceptable?

**Comment by vbekiaris** [comment]
With the latest commit, the changes in `Config.findCacheConfig`:
- do not affect the JCache implementation (it is now using `findCacheConfigOrNull` which preserves the previous behavior of `findCacheConfig`)
- align `findCacheConfig`'s behavior with the rest `findXConfig` methods -> fewer surprises in the future.

I am OK with this, as long as the change in `findCacheConfig` behavior is documented in release notes (which shouldn't affect that many users as I guess most users need a mutable config object anyway). @mmedenjak wdyt?

**Pull request merged**